extensions/panache/hibernate-orm-panache-common/runtime/src/main/java/io/quarkus/hibernate/orm/panache/common/runtime/CommonPanacheQueryImpl.java

```diff
@@ -103,7 +110,30 @@ private CommonPanacheQueryImpl(CommonPanacheQueryImpl<?> previousQuery, String n
         }
         select.append(") ");
 
-        return new CommonPanacheQueryImpl<>(this, select.toString() + query, "select count(*) " + query);
+        return new CommonPanacheQueryImpl<>(this, select.toString() + query, "select count(*) " + query, null);
+    }
+
+    public <T> CommonPanacheQueryImpl<T> select(List<String> columns, Function<Object, T> resultMapper) {
```

Comment:
Shouldn't it be `select(List<String> columns, Function<Entity, T> resultMapper)` ?

Reply:
I don't think so. We don't have to return a list of `Entity`... it could be anything, based on the mapper function.

Reply:
OK, I just understood it, it maps from an Object (the row from the resultset) to a new return type `T`.